### ISSUE TITLE
Warning about binding on address other than 127.0.0.1

### ISSUE THREAD

**Issue by mattyw** [docker/docker#10728]
This warning:
https://github.com/docker/docker/blob/master/api/server/server.go#L1558
Also gets printed if I bind to localhost - surely this shouldn't be the case?

**Comment by jfrazelle** [comment]
RIght, I will label as a bug and this would be an easy fix for anyone that
wants to contribute.

**Comment by cpuguy83** [comment]
It needs to be warning no matter what it's binding to if --tlsverify isn't set.

**Comment by phemmer** [comment]
> It needs to be warning no matter what it's binding to if --tlsverify isn't set.

Somewhat agree with this. I think it should either be always warn if binding to TCP, or never warn.

Agree because it enables docker (and thus root) access to anyone with access to the box. Disagree because I think this warning is the responsibility of documentation, not runtime (though this isn't a strong opinion).

**Comment by cpuguy83** [comment]
Generally I'm against warnings on daemons since they will probably go unnoticed, but I think this one is so important.

**Comment by jfrazelle** [comment]
But he's trying with localhost ;) we shouldn't just check against 127.0.0.1

**Comment by jfrazelle** [comment]
Either we need to be consistent or we need to take out the check for
127.0.0.1

**Comment by cpuguy83** [comment]
I see that, just making the comment that if we are going to change it, let's change it to really warn people, because localhost/127.0.0.1 is **not** safe.

**Comment by tiborvass** [comment]
@jfrazelle we do proper check for insecure registries, we check for 127/8 IPs, we could reuse/refactor the code to do the same for the warning if we want to.

**Comment by tiborvass** [comment]
And by IPs, i meant, after resolving the hostname.

**Comment by jfrazelle** [comment]
Then let's remove the check for 127.0.0.1 so it's consistent ;)

**Comment by mattyw** [comment]
Just to make sure I've kept up: We're now proposing:
```
if !job.GetenvBool("TlsVerify") {
  log.Infof("/!\\ DON'T BIND ON ANY IP ADDRESS WITHOUT setting -tlsverify IF YOU DON'T KNOW WHAT YOU'RE DOING /!\\")
	}
```

**Comment by LK4D4** [comment]
Closed by #10730
